react-app/src/components/Opensearch/main/Filtering/Drawer/index.test.tsx

```diff
@@ -21,6 +22,23 @@ const setup = (
 };
 
 describe("OsFilterDrawer", () => {
+  beforeEach(() => {
+    global.localStorage = {
+      setItem: vi.fn(),
+      getItem: vi.fn(),
+      removeItem: vi.fn(),
+      key: vi.fn(),
+      clear: vi.fn(),
+      length: 0,
+    };
+
+    vi.spyOn(global.localStorage, "getItem").mockReturnValue(null);
```

Comment:
if you store this spy in variable you wouldn't have to recreate it to change the value, for instance
```const localStorageSpy = vi.spyOn(global.localStorage, "getItem").mockReturnValue(null);``` but you would need to move it above the beforeEach